### ISSUE TITLE
Nemo crashes/traps with quick image and window/directory(?) updates

### ISSUE THREAD

**Issue by JPeisach** [linuxmint/nemo#2717]
```
 * Nemo version 4.8.6
 * Windowed Nemo
 * Distribution - Ubuntu MATE 21.04 (Alpha/Beta), installed Ubuntu Cinnamon Remix 21.04 environment
 * Intel i5-7200U CPU, Intel HD Graphics 620 GPU
 * 64 bit
 ```

**Issue**
While working on an icon theme, and manually replacing images going through the folders, nemo crashes. I got this twice-one was a trap, another was a segfault.

```
Mar 31 16:21:35 Joshua-PC kernel: [  574.248846] nemo[5129]: segfault at 5de ip 00007f0ab9562c60 sp 00007ffe4068fe48 error 4 in libgobject-2.0.so.0.6705.0[7f0ab953b000+30000]
Mar 31 16:25:05 Joshua-PC kernel: [  784.271992] traps: nemo[7460] trap stack segment ip:7f9eaa572db7 sp:7ffceefa9f40 error:0 in libgtk-3.so.0.2404.21[7f9eaa1fc000+385000]
```

addr2line first crash:
```
jpeisach@Joshua-PC ~> addr2line -e /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.6705.0 27C60 -fCi
g_signal_handler_disconnect
??:?
```

**
addr2line second crash (This is the most likely cause)
```
jpeisach@Joshua-PC ~> addr2line -e /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2404.21 376DB7 -fCi
gtk_tree_store_move_after
??:?
```
**

(above is two different crashes caused by same issue while I was working)

**Steps to reproduce**
It just happened, quickly. I can't get an exact thing, but this is the closest idea:
1) Open nemo.. twice. Two different windows, two different directories
2) Move icons from folder into the other folder in the different window. The images must have the same name and probably symlinked.
3) Replace the images for all of them
4) Whenever doing some action, like switching out of the window, or maybe Alt+Tabing or going to another folder by going back a directory, nemo will crash/trap.
Hopefully a clear reason will be found for this on reproducing.

**Expected behaviour**
Nemo won't crash.

**Other information**
This is an Alpha/Beta build and I actually haven't updated the system yet.
No nemo extensions installed
https://termbin.com/b8he - syslog
https://termbin.com/wfpt - xsession errors displays failures at times of crashes
https://termbin.com/iqi5 - crash dump

If more information is needed, for when this (hopefully) gets backported to Ubuntu, see the launchpad bug:
https://bugs.launchpad.net/ubuntu/+source/nemo/+bug/1922129

**Comment by ItzSwirlz** [comment]
> It only happened once?  Do you have a coredump?
> 

No. Not that I could find but that's why I attached the LP bug in case Ubuntu-bug picked it up. It happened twice during my work and it's probably two different crashes but are related.

> What does alpha/beta build mean - you mean alpha iso?  Is nemo built off git master or the latest tagged release?
> 

We are at our Beta Freeze point tomorrow. Built on tagged version.

> If you can reproduce it and provide more info please - coredump, more environment info...did you check the gtk bug list for anything similar? 

I will try to reproduce. Environment info should be on the launchpad but I can termbin it. I'll check the bug list

**Comment by ItzSwirlz** [comment]
Update: Got a crash dump. https://termbin.com/iqi5
